src/pods/generate/TemplatedResourcesGenerator.ts

```diff
@@ -57,7 +58,19 @@ export class TemplatedResourcesGenerator implements ResourcesGenerator {
   public async* generate(location: ResourceIdentifier, options: Dict<string>): AsyncIterable<Resource> {
     const mapper = await this.factory.create(location.path, this.templateFolder);
     const folderLink = await this.toTemplateLink(this.templateFolder, mapper);
-    yield* this.processFolder(folderLink, mapper, options);
+    const iterable = this.processFolder(folderLink, mapper, options);
+
+    // Temporary code to trace race condition https://github.com/solid/community-server/issues/1059
+    const results: Resource[] = [];
+    for await (const resource of iterable) {
+      results.push(resource);
+    }
+    const expected = readdirSync(folderLink.filePath);
+    assert.equal(results.length, expected.length, `Bug #1059 @generate\nExpected: ${
+      expected.join()}\nActual: ${
+      JSON.stringify({ folderLink, results }, null, '  ')}`);
```

Comment:
Mentioned in the PR comments, but yea, this is not going to work since `processFolder` is a recursive function. So you would also have to do a recursive `readdirSync`.